src/rofl/common/crofsock.cc

```diff
@@ -1288,106 +1288,129 @@ void crofsock::handle_timeout(cthread &thread, uint32_t timer_id) {
   }
 }
 
-void crofsock::send_message(rofl::openflow::cofmsg *msg) {
+enum crofsock::msg_result_t crofsock::send_message(rofl::openflow::cofmsg *msg, bool enforce_queueing, bool keep_message) {
+
   VLOG(3) << __FUNCTION__ << " msg=" << msg
           << " txqueue_pending_pkts=" << txqueue_pending_pkts
           << " tx_disabled=" << tx_disabled
           << " tx_is_running=" << tx_is_running;
 
   if (tx_disabled) {
-    delete msg;
-    return;
+	if (!keep_message) {
```

Comment:
I suggest to remove the `keep_message` feature, because it is not used anywhere at the moment, it creates overhead, a user can never set it, and if it is set, it is unclear when to `delete msg;`